website/docs/pipelines/promoting-applications.mdx

```diff
@@ -130,133 +133,186 @@ Be sure to create the Provider/Alert tuple on **each of the leaf clusters
+Now as soon as the `HelmRelease` on the first environment defined in the pipeline is bumped (e.g. by Flux discovering a new version in the Helm repository), an event is sent to the promotion webhook which will determine the next action based on the pipeline definition and chosen strategy. The rest of this guide describes how to setup up any of the available strategies depending on your requirements.
 
 ## Pull request
 
+:::danger
+
+Creating pull requests requires write access to your git repo. In case your pull request secret is compromised (and you could assume
+it as a likely scenario), it could end up, for example, in deleting your production applications.
+
+Please make sure you understand [Security](#security) section below before enabling automated pull requests.
+:::
+
 This section covers adding a promotion by pull request (PR) strategy, so that whenever the application defined in a pipeline
 is upgraded in one of the pipeline's environments, a PR is created that updates the manifest file setting the application version in the next environment.
 
 The dynamic nature of GitOps deployments requires you to assist Weave GitOps a little with information on which repository hosts the manifest files,
 how to authenticate with the repository and the Git provider API, and which file hosts the version definition for each environment.
 
-:::caution
+### Security
 
-Creating pull requests requires read and write access in your git repo.
-A compromised token could lead to changing the git repository and creation of legitimate looking changes/pull requests.
+#### Environments and Repositories
 
-Ensure you understand and adopt [security recommendations](./#security-recommendations) before using the feature.
+1. Use it in low-risk environments and not in production: pipelines is an alpha feature and not yet production-ready.
+2. Make sure you have considered possible attack vectors to production, and put controls in place. Some
+of these scenarios are:
+- In the case of a [monorepo](https://fluxcd.io/flux/guides/repository-structure/#monorepo): you want to ensure that a compromised token doing a change in test cannot end up doing a promotion in production without a pull request review, for example, leveraging [Codeowners](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners).
+- In the case of [repo per environment](https://fluxcd.io/flux/guides/repository-structure/#repo-per-environment) you want to ensure that a pipeline is configured with your test environment repo URL. You also want to ensure that you have seggregation of tokens per environment rather than allowing a token to access any environment repo.
```

Comment:
```suggestion
- In the case of [repo per environment](https://fluxcd.io/flux/guides/repository-structure/#repo-per-environment) you want to ensure that a pipeline is configured with your test environment repo URL. You also want to ensure that you have segregation of tokens per environment rather than allowing a token to access any environment repo.
```